pytorch_blade/src/compiler/mlir/runtime/ral_context.cpp

```diff
@@ -197,11 +197,29 @@ torch::List<torch::Tensor> RalContext::CreateAndBindingOutputs(
     auto option = torch::device(dev_type)
                       .dtype(scalar_type)
                       .memory_format(torch::MemoryFormat::Contiguous);
-    torch::Tensor out_tensor = IsEmptyTensor(out_buf->shape())
-        ? torch::zeros(out_buf->shape(), option)
-        : torch::from_blob(
+    torch::Tensor out_tensor;
+    if (IsEmptyTensor(out_buf->shape())) {
+      out_tensor = torch::zeros(out_buf->shape(), option);
+    } else if (out_buf->owned()) {
+#ifdef TORCH_BLADE_BUILD_WITH_CUDA
+      auto deleter = (output_info.device == "cuda")
+          ? c10::cuda::CUDACachingAllocator::raw_delete
+          : c10::free_cpu;
+#else
+      auto deleter = c10::free_cpu;
+#endif
+      out_tensor = torch::from_blob(
+          const_cast<void*>(out_buf->data()),
+          out_buf->shape(),
+          deleter,
+          option);
+      out_buf->release();
+    } else {
+      out_tensor =
```

Comment:
Nit: Shall we decrease the number of buffer users in this case? And make the final user reuse the underline buffer